init.c

```diff
@@ -55,14 +55,18 @@
 #include <sys/wait.h>
 #include <sys/time.h>
 
-#define CHECK_PAGER \
-  if ((CurrentMenu == MENU_PAGER) && (idx >= 0) &&	\
-	    (MuttVars[idx].flags & R_RESORT)) \
-	{ \
-	  snprintf (err->data, err->dsize, \
-	    _("Not available in this menu.")); \
-	  return (-1); \
-	}
+static int check_pager(int idx, BUFFER *err)
+{
+  if ((CurrentMenu == MENU_PAGER) && (idx >= 0) && (MuttVars[idx].flags & R_RESORT))
+  {
+    snprintf(err->data, err->dsize, _("Not available in this menu."));
+    return (-1);
+  }
+  else
+  {
+    return 0;
+  }
+}
 
```

Comment:
what exactly do you mean by "caller" ? do you mean with that the surrounding function that calls somewhere in its body check_pager() ?